frame/contracts/src/lib.rs

```diff
@@ -861,11 +858,7 @@ pub mod pallet {
 		Indeterministic,
 	}
 
-	/// A mapping from an original code hash to the original code, untouched by instrumentation.
-	#[pallet::storage]
-	pub(crate) type PristineCode<T: Config> = StorageMap<_, Identity, CodeHash<T>, CodeVec<T>>;
-
-	/// A mapping between an original code hash and instrumented wasm code, ready for execution.
+	/// A mapping between an original code hash and Wasm code, ready for execution.
 	#[pallet::storage]
 	pub(crate) type CodeStorage<T: Config> =
 		StorageMap<_, Identity, CodeHash<T>, PrefabWasmModule<T>>;
```

Comment:
> Min und max memory does not need to be stored AFAIK. We can just query wasmi for it when loading the module.

No we can't query wasmi for that, because it does not provide any method for that. You can get [Pages](https://docs.rs/wasmi/latest/wasmi/core/struct.Pages.html) but not the inner field which is private

Reply:
> No we can't query wasmi for that, because it does not provide any method for that. You can get [Pages](https://docs.rs/wasmi/latest/wasmi/core/struct.Pages.html) but not the inner field which is private.

Parsing it with `parity-wasm` every time we execute a contract [decreases](https://github.com/paritytech/substrate/commit/d3f59b2e4d850ccddf1eb523b2fb547906748eff#diff-ffebabc685eaf81fc15f476b128bc403c9db3d8fd5d36a00de831cfe1c850293R2125) performance significantly.
Hence for now it seems the right solution is to keep it in storage.

Reply:
> No we can't query wasmi for that, because it does not provide any method for that. You can get [Pages](https://docs.rs/wasmi/latest/wasmi/core/struct.Pages.html) but not the inner field which is private.

Okay we can do a workaround by using [`wasmi::core::Pages::to_bytes()`](https://docs.rs/wasmi/0.28.0/wasmi/core/struct.Pages.html#method.to_bytes) for that. Or wait untill wasmi provides a nicer getter for that. Both could be done in a follow-up PR.

Reply:
I think you should do it in this PR in order to get rid of the fields that are storing the pages. We can just query it whenever we use the module. Let's get rid of all the caches!

Reply:
done